Samples/Samples/ViewModel/MediaPickerViewModel.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Windows.Input;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace Samples.ViewModel
+{
+    public class MediaPickerViewModel : BaseViewModel
+    {
+        string mediaPath;
+
+        public MediaPickerViewModel()
+        {
+            PickPhotoCommand = new Command(DoPickPhoto);
+
+            MediaPicker.MediaPicked += OnMediaPicked;
+        }
+
+        public ICommand PickPhotoCommand { get; }
+
+        public string MediaPath
+        {
+            get => mediaPath;
+            set => SetProperty(ref mediaPath, value);
+        }
+
+        async void DoPickPhoto()
+        {
+            try
+            {
+                var photo = await MediaPicker.ShowPhotoPickerAsync();
+
+                // canceled
+                if (photo == null)
+                {
+                    MediaPath = null;
+                    return;
+                }
+
+                // save the file into local storage
+                var newFile = Path.Combine(FileSystem.CacheDirectory, photo.FileName);
+                using (var stream = await photo.OpenReadAsync())
+                using (var newStream = File.OpenWrite(newFile))
+                {
+                    await stream.CopyToAsync(newStream);
+                }
```

Comment:
This is the not-too-heavy code to copy the picked image to a local folder for viewing. We could add helpers, such as `MediaFile.CopyAsync(Stream)` and `MediaFile.CopyAsync(string)`.